dask/array/tests/test_reductions.py

```diff
@@ -902,3 +902,33 @@ def w_sum(x, weights=None, dtype=None, computing_meta=False, **kwargs):
     # Non-broadcastable weights (too many dims)
     with pytest.raises(ValueError):
         da.reduction(dx, w_sum, np.sum, weights=[[[2]]])
+
+
+def test_ravel_by_chunks():
+    x = np.random.randint(10, size=(4, 6))
+
+    # 2d
+    for chunks in [(4, 6), (2, 6)]:
+        a = da.from_array(x, chunks=chunks)
+        assert len(a.ravel_chunks().dask) == len(a.dask) + len(a.chunks[0])
+        assert np.sum(x.ravel().shape) == np.sum(a.ravel_chunks().compute().shape)
```

Comment:
```suggestion
        assert np.sum(x.ravel().shape) == np.sum(a.ravel_chunks().shape)
```

Reply:
I suggested a bunch of changes like this. We can test that the computed shape is correct, but we also want to make sure the pre-computed `shape` is correct.